### ISSUE TITLE
LambdaRank の二値ラベル化（relevanceスコア → 3着以内の二値ラベル）

### ISSUE THREAD

**Issue by maruyamamtk** [maruyamamtk/keiba_prediction#85]
## 概要

現在のシステムは LightGBM LambdaRank で着順ベースの4段階 relevance スコア（1着=3, 2着=2, 3着=1, 4着以下=0）を使用しているが、これを **3着以内=1, それ以外=0 の二値ラベル**に変更する。

### 意図
- **ランク学習の枠組みは維持する**（objective=lambdarank, レース単位のグループ学習）
- ラベルのみを二値に変更し、「3着以内に入るかどうか」をランク学習で最適化する
- LambdaRank の「レース内の相対順位を学習する」特性と、「複勝圏内（3着以内）かどうか」の明確な目的を組み合わせる

### 変更しないもの
- `objective: lambdarank`（変更しない）
- `metric: ndcg`（変更しない）
- `group` パラメータ（レース単位のグループ学習は維持）
- `LGBMRanker` クラス（そのまま使用）

## ブランチ

`feature/binary-classification-model`

### 修正
1. **`src/models/train.py`** - `prepare_features()` のラベル生成ロジックを二値に変更
2. **`config/model_config.yaml`** - コメント修正（ラベル定義の説明更新）
3. **`tests/test_train.py`** - ラベルのアサーションを二値に更新
4. **`tests/test_predict.py`** - ラベル生成部分の更新

### 変更なし
- `src/models/lgbm_ranker.py` - 変更不要（LambdaRank のまま）
- `tests/test_lgbm_ranker.py` - 変更不要
- `src/models/predict.py` - 変更不要（スコアの大小関係は同じ）
- `src/ml/features/feature_query_raw.sql` - 変更不要

### 新規作成なし
- `lgbm_classifier.py` は不要（LGBMRanker をそのまま使用）

#### `prepare_features()` のラベル変更
```python
# 変更前: 4段階 relevance スコア
positions = df["finish_position"].values.astype(int)
y = np.where(positions == 1, 3,
     np.where(positions == 2, 2,
      np.where(positions == 3, 1, 0)))

# 変更後: 二値ラベル（3着以内=1, それ以外=0）
positions = df["finish_position"].values.astype(int)
y = np.where((positions >= 1) & (positions <= 3), 1, 0)
```
- `finish_position=0`（出走取消等）は 0 として扱う（`positions >= 1` の条件で対策）
- docstring を更新し、ラベル定義の変更を明記

#### `evaluate_predictions()` の修正
- 既存の NDCG@3 / Recall@3 はそのまま維持（ランク学習の評価として有効）
- 追加指標として **AUC**（二値ラベルの予測精度）を計算
```python
from sklearn.metrics import roc_auc_score

# レース横断でのAUC
binary_labels = np.where((y_true_positions >= 1) & (y_true_positions <= 3), 1, 0)
auc = roc_auc_score(binary_labels, y_pred)
```

### 2. `config/model_config.yaml`（修正）
- ファイル先頭のコメントを更新: 「4段階 relevance」→「二値ラベル（3着以内=1, 他=0）」
- `params` セクションは変更なし（lambdarank のまま）
- `evaluation.metrics` に `auc` を追加

#### `TestPrepareFeatures` の更新
```python
# 変更前
assert y[0] == 3  # 1着 → 3
assert y[1] == 2  # 2着 → 2
assert y[2] == 1  # 3着 → 1

# 変更後
assert y[0] == 1  # 1着 → 1（3着以内）
assert y[1] == 1  # 2着 → 1（3着以内）
assert y[2] == 1  # 3着 → 1（3着以内）
```

#### `TestEvaluatePredictions` の更新
- AUC 指標が結果に含まれることを検証

#### `test_zero_finish_position` の更新
```python
# 0着 → 0（3着以内でない）
assert y[0] == 0
# 1着 → 1（3着以内）
assert y[1] == 1
```

### 4. `tests/test_predict.py`（修正）
- `trained_model_path` fixture 内のラベル生成を二値に更新

## 実装順序
1. ブランチ作成
2. `train.py` の `prepare_features()` ラベル変更
3. `train.py` の `evaluate_predictions()` に AUC 追加
4. `config/model_config.yaml` のコメント・評価指標更新
5. テスト更新
6. 全テスト実行・確認

## 検証方法
```bash
python -m pytest tests/test_train.py tests/test_predict.py tests/test_lgbm_ranker.py -v
```

## 影響範囲
- **小さい変更**: ラベル生成ロジック1箇所の変更が主。LGBMRanker クラスや predict.py は変更不要
- **後方互換性**: 出力スコアの大小関係は変わらないため、predict.py の pred_score / pred_rank には影響なし
- **リスク**: 二値ラベルにすることで NDCG の計算が「1着と2着と3着を区別しない」形になるが、目的（複勝圏内予測）に合致

## 関連Issue
- #86: Optunaによるハイパーパラメータ調整（本Issueの後に実施）
